mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/ExtensionModelFactory.java

```diff
@@ -175,7 +174,6 @@ public final class ExtensionModelFactory {
   public ExtensionModelFactory() {
     declarationEnrichers = unmodifiableList(asList(
                                                    new ParameterDslDeclarationEnricher(),
-                                                   new BooleanParameterDeclarationEnricher(),
```

Comment:
let's take a different approach. The reality is that this should always apply as it enforces a language rule. The fact that XML SDK breaks is a bug. Also, we do want this enforced in reuse and Mule SDK use cases, and not all crafted extensions will have an issue with this.

Let problematic extensions set a flag in the ExtensionLoadingContext so that the enricher skips.